pyEpiabm/pyEpiabm/core/cell.py

```diff
@@ -70,16 +71,24 @@ def add_microcells(self, n):
         for _ in range(n):
             self.microcells.append(Microcell(self))
 
-    def set_id(self, id: float):
+    def set_id(self, id: str):
         """Updates ID of cell (i.e. for input from file).
 
         Parameters
         ----------
-        id : float
+        id : str
             Identity of cell
 
         """
-        self.id = id
+        # Ensure id is a string
+        if not isinstance(id, str):
+            raise TypeError("id must be of type string")
+
+        # May want to set upper limit on the number of digits
```

Comment:
Really comment should match the code. We occasionally allow TODO comments, but I don't think this merits that (should be for changes that resolve active problems) - if you think this is important enough to change then either do so in this PR, or make an issue for it.

Reply:
Instead - I would document the regex as this is not readable for everyone